backend/src/main/java/ai/verta/modeldb/configuration/AppConfigBeans.java

```diff
@@ -0,0 +1,412 @@
+          "ArtifactStore service initialized and resolved storage dependency before server start");
+      return artifactStoreService;
+    } else {
+      LOGGER.info("Artifact store service is disabled.");
+      return null;
+    }
+  }
+
+  public static boolean migrate(MDBConfig mdbConfig, JdbiUtil jdbiUtil, Executor handleExecutor)
+      throws Exception {
+    var liquibaseMigration =
+        Boolean.parseBoolean(
+            Optional.ofNullable(System.getenv(CommonConstants.ENABLE_LIQUIBASE_MIGRATION_ENV_VAR))
+                .orElse("false"));
+
+    var modelDBHibernateUtil = ModelDBHibernateUtil.getInstance();
+    modelDBHibernateUtil.initializedConfigAndDatabase(mdbConfig, mdbConfig.getDatabase());
+
+    if (liquibaseMigration) {
+      LOGGER.info("Liquibase migration starting");
+      jdbiUtil.runLiquibaseMigration();
+      LOGGER.info("Liquibase migration done");
+
+      modelDBHibernateUtil.createOrGetSessionFactory(mdbConfig.getDatabase());
+      LOGGER.info("Old code migration using hibernate starting");
+      modelDBHibernateUtil.runMigration(mdbConfig.getDatabase(), mdbConfig.migrations);
+      LOGGER.info("Old code migration using hibernate done");
+
+      LOGGER.info("Code migration starting");
+      jdbiUtil.runMigration(handleExecutor);
+      LOGGER.info("Code migration done");
+
+      boolean runLiquibaseSeparate =
+          Boolean.parseBoolean(
+              Optional.ofNullable(System.getenv(CommonConstants.RUN_LIQUIBASE_SEPARATE))
+                  .orElse("false"));
+      LOGGER.trace("run Liquibase separate: {}", runLiquibaseSeparate);
+      if (runLiquibaseSeparate) {
+        return true;
+      }
+    }
+
+    modelDBHibernateUtil.createOrGetSessionFactory(mdbConfig.getDatabase());
+    jdbiUtil.setIsReady();
+
+    return false;
+  }
+
+  @Bean
+  public CommandLineRunner commandLineRunner(ApplicationContext ctx) {
+    return args -> {
+      try {
+        LOGGER.info("Backend server starting.");
+        final var logger =
+            java.util.logging.Logger.getLogger("io.grpc.netty.NettyServerTransport.connections");
+        logger.setLevel(Level.WARNING);
+
+        final var config = ctx.getBean(MDBConfig.class);
+
+        // Configure server
+        final var services = ctx.getBean(ServiceSet.class);
+        final var jdbiUtil = ctx.getBean(JdbiUtil.class);
+
+        // Initialize executor so we don't lose context using Futures
+        final var handleExecutor = ctx.getBean(Executor.class);
+
+        // Initialize database configuration and maybe run migration
+        if (migrate(config, jdbiUtil, handleExecutor)) {
+          LOGGER.info("Migrations have completed.  System exiting.");
+          appContext.initiateShutdown(0);
+          return;
+        }
+
+        LOGGER.info("Migration disabled, starting server.");
+        final var daos = ctx.getBean(DAOSet.class);
+
+        // Initialize telemetry
+        initializeTelemetryBasedOnConfig(config);
```

Comment:
shouldn't this just be another bean definition?

Reply:
Resolved